lib/ash/actions/read/read.ex

```diff
@@ -151,6 +152,16 @@ defmodule Ash.Actions.Read do
 
     query =
       if opts[:initial_data] do
+        query =
+          query
+          |> Ash.Query.set_context(%{
```

Comment:
If i interpret this code correctly https://github.com/ash-project/ash/blob/ff1a590f46e750422920b4dc200fc6d3329b420c/lib/ash/actions/read.ex#L369 it would take the pagination values configured in the action if there were none in opts. 

But I only have a superficial understanding of what exactly is happening there

Reply:
That logic exists in `page_opts/2`, so we should be good.